test/xray.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+const sinon = require('sinon');
+const { expect } = require('chai');
+const AWSXRay = require('aws-xray-sdk');
+
+const XRay = require('../lib/xray');
+
+describe('XRay', function () {
+  beforeEach(function () {
+    this.asyncFunctionWrapper = XRay.asyncFunctionWrapper;
+    this.log = {
+      warn: Function.prototype
+    };
+    sinon.stub(this.log, 'warn');
+    this.resolve = sinon.stub().resolves();
+    this.reject = sinon.stub().rejects();
+    this.context = {
+      log: this.log
+    };
+  });
+
+  afterEach(function() {
+    sinon.verifyAndRestore();
+  });
+
+  describe('trace', function () {
```

Comment:
These tests do not test `trace` function (except the first two), thus should be split into multiple `describe` blocks.